sdk/tdf.go

```diff
@@ -664,6 +666,31 @@ func (s SDK) LoadTDF(reader io.ReadSeeker, opts ...TDFReaderOption) (*Reader, er
 		return nil, fmt.Errorf("newAssertionConfig failed: %w", err)
 	}
 
+	if config.kasAllowlist == nil && !config.ignoreAllowList { //nolint:nestif // need to handle both cases
+		if s.KeyAccessServerRegistry != nil {
+			// retrieve the registered kases if not provided
+			kases, err := s.KeyAccessServerRegistry.ListKeyAccessServers(context.Background(), &kasregistry.ListKeyAccessServersRequest{})
+			if err != nil {
+				return nil, fmt.Errorf("kasregistry.ListKeyAccessServers failed: %w", err)
+			}
+			config.kasAllowlist = AllowList{}
+			for _, kas := range kases.GetKeyAccessServers() {
+				err = config.kasAllowlist.Add(kas.GetUri())
+				if err != nil {
+					return nil, fmt.Errorf("kasAllowlist.Add failed: %w", err)
+				}
+			}
+			// grpc target does not have a scheme
+			err = config.kasAllowlist.Add("http://" + s.conn.Target())
```

Comment:
Should we default to `https`?

Reply:
this is basically just adding a scheme so the url parsing works correctly, its removed when its actually added to the allowlist, it doesnt matter what the scheme is here